src-tauri/src/main.rs

```diff
@@ -156,26 +153,42 @@ async fn main() {
                 // and move it to another function or thread
                 let item_handle = app.tray_handle().get_item(&id);
                 match id.as_str() {
-                    "hide" => {
-                        let window = app.get_window("main").unwrap();
-                        window.hide().unwrap();
+                    "hide_show" => {
+                        let window = app.get_window("main").expect("Couldn't fetch window");;
+
                         // you can also `set_selected`, `set_enabled` and `set_native_image` (macOS only).
-                        item_handle.set_title("Show").unwrap();
+                        match window.is_visible().expect("Couldn't get window visibility") {
+                            true => {
+                                window.hide().expect("Couldn't hide window");
+                                item_handle.set_title("Show").expect("Couldn't change system tray item to show")
+                            }
+                            false => {
+                                window.show().expect("Couldn't show window");
+                                item_handle.set_title("Hide").expect("Couldn't change system tray item to hide")
+                            }
+                        }
+
+                        window.clone().on_window_event(move |window_event| {
+                            if let WindowEvent::CloseRequested { .. } = window_event {
+                                window.hide().expect("Couldn't hide window");
+                                item_handle.set_title("Show").expect("Couldn't change system tray item to show");
+                            }
+                        })
                     }
                     "quit" => {
                         app.exit(0);
                     }
-                    "show" => {
-                        let window = app.get_window("main").unwrap();
-                        window.show().unwrap();
-                        // you can also `set_selected`, `set_enabled` and `set_native_image` (macOS only).
-                    }
+
                     _ => {}
                 }
             }
             SystemTrayEvent::LeftClick { .. } => {
                 let window = app.get_window("main").unwrap();
                 window.show().unwrap();
+                app.tray_handle()
+                    .get_item("hide_show")
+                    .set_title("Hide")
+                    .unwrap();
```

Comment:
Would be good to convert these unwraps to expect as well

Reply:
Fixed in: b074368